### PR TITLE
Pull git submodules in the server's main

### PR DESCRIPTION
**Pull request by vladsaveliev**
That wasn't the right place to pull submodules in the previous PR :) (though still useful to keep).

### CODE REVIEW

File: server/main.py [hunk 1]
```diff
@@ -162,6 +162,8 @@ async def index(request):
             job.command('git checkout main')
             job.command(f'git merge-base --is-ancestor {quote(commit)} HEAD')
         job.command(f'git checkout {quote(commit)}')
+        job.command(f'git submodule init')
```
Comment (lgruen) [inline server/main.py:165]: Can we do a `--recurse-submodules` in the `clone` step instead?

**Review by lgruen — CHANGES_REQUESTED** [review]

File: server/main.py (hunk 1 above)
Comment (vladsaveliev) [reply server/main.py:165]: Don't think that would work if the submodule exists only in a branch, but not in main

Comment (vladsaveliev) [reply server/main.py:165]: Only with a subsequent `git submodule update`

Comment (vladsaveliev) [reply server/main.py]: Updated!

**Review by lgruen — APPROVED** [review]
Nice!

**Pull request merged**